### PR TITLE
feat: add uv and update github actions

### PR DESCRIPTION
**Pull request by niltonpimentel02**
@bbelderbos as expected... errors 😆 tomorrow I will try again!

### CODE REVIEW

File: .github/workflows/python-app.yml [hunk 1]
```diff
@@ -11,35 +11,29 @@ on:
...
+      - uses: actions/checkout@v4
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
+
+      - name: Set up Python
+        run: uv python install 3.9
+
+      - name: Install the project
+        run: uv pip sync requirements.txt
```
Comment (bbelderbos) [inline .github/workflows/python-app.yml]: if we switch to pyproject.toml we can use just `uv sync` I expect

File: .github/workflows/python-app.yml [hunk 2]
```diff
@@ -11,35 +11,29 @@ on:
...
+
+      - name: Run tests
+        # For example, using `pytest`
+        run: uv run pytest tests
+        env:
+          SECRET_KEY: 21290asdkja
+          DEBUG: True
+          DATABASE_URL: sqlite:///test.db
```
Comment (bbelderbos) [inline .github/workflows/python-app.yml:32]: hm, can we use an in-memory sqlite db?

File: .github/workflows/python-app.yml [hunk 3]
```diff
@@ -11,35 +11,29 @@ on:
...
+
+      - name: Set up Python
+        run: uv python install 3.9
+
+      - name: Install the project
+        run: uv pip sync requirements.txt
+
+      - name: Run tests
+        # For example, using `pytest`
+        run: uv run pytest tests
```
Comment (bbelderbos) [inline .github/workflows/python-app.yml]: what about just `uv run pytest`?

File: .github/workflows/python-app.yml [hunk 4]
```diff
@@ -11,35 +11,29 @@ on:
...
-        FROM_EMAIL: pybites@example.com
-        ADMIN_USERS: pybites
-        CELERY_BROKER_URL: amqp://guest:guest@localhost
+      - uses: actions/checkout@v4
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
+
+      - name: Set up Python
+        run: uv python install 3.9
```
Comment (bbelderbos) [inline .github/workflows/python-app.yml]: can we use a more recent one? I am happy to go with >= 3.12

File: .github/workflows/python-app.yml (hunk 2 above)
Comment (niltonpimentel02) [reply .github/workflows/python-app.yml:32]: no idea

**Comment by niltonpimentel02** [comment]
@bbelderbos I think we've made it! 🎉

**Review by bbelderbos — APPROVED** [review]
Nice, thank you!

**Comment by bbelderbos** [comment]
I think we should ditch develop branch (https://github.com/PyBites-Open-Source/pybitesbooks/issues/82), what do you think?

**Pull request merged**